### DISCUSSION TITLE
Splitting Issue User Sign-In/log-in #12 into 2 separate task.

### DISCUSSION THREAD

**Discussion by MiguelGibo** [uprm-inso4101-2024-2025-s1/semester-project-campus-resource-and-event-management#36] [Ideas]
**Reason:** The task involves too many issues, making completion criteria hard to identify. It would be best handled if split up into smaller, more reasonable task.

**Suggested Solution:** Making two new tasks, which would be roughly outlined as followed.

### Task 1:
**Tittle:** User Sign-up Implementation
**Objective:** Implement a user page that would allow new users to create and populate a new account.

**Summary:** To be provided access to our platform, new users need the ability to create and populate an account. This feature will enable users to sign up, provide necessary information, and gain access to manage their reservations or event listings.

**Proposed Solution:** 
- **Initial Prompt:** When users first access the platform, they should be prompted to either sign up for a new account or log in if they already have one.
- **Sign-Up Requirements:** The platform should prompt the user to provide essential information for account creation:
  - Full Name (First Name, Middle Name, Last Names)
  - Email Address
  - Password
  - Student Number (if applicable)
  - Additional fields as needed (e.g., phone number, department, role). Suggestions for more fields are welcome.
  - Validate all inputs to ensure data integrity (e.g., valid email format, password strength).
  - Once validated, create and store the new user account securely in the database.
  - After successful account creation, display a confirmation message to the user and automatically log them in.

**Error Handling:** 
- Prevent duplicate accounts by checking for existing credentials (e.g., duplicate email or student number) during the sign-up process.
- Provide real-time feedback for invalid inputs (e.g., invalid email format, weak password) and guide users to correct errors.

**Additional Context:**
- For the time being, it might be best to implement this in a mock system. As suggested in #2 
- Security measures, such as hashing passwords, email verification, and CAPTCHA, should be planned and implemented in separate tasks.
- As discussed in issue #34, consider limiting sign-ups to preapproved university email domains (e.g., @upr.edu, @uprm.edu) to maintain user eligibility.

### Task 2:
**Tittle:** User Log-In Implementation
**Objective:** Enable existing users to log in to their accounts to manage event reservations and listings.

**Summary:** To provide users with access to their profiles and manage their reservations and event listings, we need to implement a user log-in functionality that allows them to securely access their accounts.

**Proposed Solution:** 
- **Initial Prompt:** When users access the platform, they should be prompted to either sign up for a new account or log in if they already have one.
- **Log-in Requirements:** 
  - Develop a "Log-In" page where users can enter their credentials (Email and Password) to access their accounts.
  - Add options for "Forgot Password" to help users reset their passwords. (Password recovery system should be handled in a separate task. Implement interface, but leave nonfunctional for now.)
- **Authentication:** Verify the user's email and password against the stored credentials in the database.
- **Session Management:** Upon successful authentication, establish a user session and redirect the user to their profile page or homepage. (A "Remember Me" functionality, to keep a user logged in on a specific device, may be a good task to implement in the future.)

**Error Handling:** 
- Show user an error messages for incorrect email or password and account not found.
- Prompt user with option to "Forgot password" and "Sign up" options if a failed attempt is made.

**Additional Context:**
- For the time being, it might be best to implement this in a mock system. As suggested in #2

**Comment by AngelUPR** [comment]
Issue has been created from discussion into two different issues
